### ISSUE TITLE
Share projects without access to source code?

### ISSUE THREAD

**Issue by brollb** [NetsBlox/NetsBlox#3447]
When used in the classroom, sometimes it makes sense to share a project without access to the source (ie, the solution) to structured activities or projects. One easy solution to this would be to make another project (like the gallery) that embeds the project in an iframe.

However, a few other questions should probably be answered before diving in:
- What does the workflow look like for a teacher trying to share a project solution?
- Are there any other features that should be included in an initial version? Project description?
- Should the embedded URL contain the XML or a link to a public project?

**Comment by gsteinLTU** [comment]
This sounds like it should mostly change things client-side with a flag set when sharing the project that disables things like the code tab/export/etc while the project is open. 

>Should the embedded URL contain the XML or a link to a public project?

I think the public project makes more sense, since these would almost always belong to an account and if the teacher needs to make a change to fix something after sharing it, they're still able to without worrying if students have the old link or not.

**Comment by brollb** [comment]
> This sounds like it should mostly change things client-side with a flag set when sharing the project that disables things like the code tab/export/etc while the project is open.

Yeah, technically there is the `embedMode` option but this is easy to bypass by just removing the flag from the query string.

> if the teacher needs to make a change to fix something after sharing it, they're still able to without worrying if students have the old link or not (or having to edit some gallery-type page as well as their project)

Good point :+1:
